stablehlo/dialect/TypeInference.cpp

```diff
@@ -1443,8 +1443,83 @@ LogicalResult inferCholeskyOp(
     return emitOptionalError(
         location, "minor dimensions of 'a' must have equal size, got shape ",
         aShape, ".");
-  inferredReturnShapes.emplace_back(aRankedType.getShape(),
-                                    aRankedType.getElementType());
```

Comment:
Similarly to #885, I'd like to propose to update the PR to just `emplace_back` the components of `aRankedType`. Unfortunately `emplace_back(aRankedType)` won't work because it doesn't account for encoding, so we'll need to explicitly call out the three components.